jobboard/tests.py

```diff
@@ -182,3 +186,57 @@ def test_page_display_to_user_that_is_recruiter(client):
     jobs_to_display = Job.get_jobs_posted_on_or_after_specific_date(date_six_months_ago)
     # test that jobs displayed are only jobs created 6 months ago
     assert response.context['jobs_to_display'] == jobs_to_display
+
+
+@pytest.fixture
+def example_recruiter():
+    example_user = User.objects.create_user("test", "testpassword")
+    example_user.save()
+    recruiter = Recruiter(user=example_user, name="test_recruiter",
+                          company=Company.objects.get(name="example_company_a"), email="test@email",
+                          phone_number="test_phone")
+    recruiter.save()
+    return recruiter
+
+
+@pytest.fixture
+def example_job_data_as_dictionary():
+    example_job = {"title": 'example_job',
+                   "major": student.models.Major.COMPUTER_SCIENCE,
+                   "job_type": 'Full-Time',
+                   "work_from": 'Office-Only',
+                   "description": 'Junior software engineer',
+                   "city": jobboard.models.City.objects.get(name="Tel Aviv-Yafo").id,
+                   "address": 'Hashalom',
+                   "title_keywords": [jobboard.models.JobTitleKeyword.objects.get(keyword="software").id]}
+    return example_job
+
+
+@pytest.mark.django_db
+def test_create_new_job_form_with_valid_data(example_recruiter, example_job_data_as_dictionary,
+                                             client):
+    client.force_login(example_recruiter.user)
+    response = client.post("/recruiter/create_new_job_form/", data=example_job_data_as_dictionary)
+    assert response.status_code == 302
+    assert response.url == "/job_created_successfully/"
+    example_job_from_db = Job.objects.get(title=example_job_data_as_dictionary["title"])
+    assert example_job_from_db.title == example_job_data_as_dictionary['title']
+    assert example_job_from_db.major == example_job_data_as_dictionary['major']
+    assert example_job_from_db.job_type == example_job_data_as_dictionary['job_type']
+    assert example_job_from_db.work_from == example_job_data_as_dictionary['work_from']
+    assert example_job_from_db.description == example_job_data_as_dictionary['description']
+    assert example_job_from_db.city.id == example_job_data_as_dictionary['city']
+    assert example_job_from_db.address == example_job_data_as_dictionary['address']
+    assert [keyword.id for keyword in example_job_from_db.title_keywords.all()] == example_job_data_as_dictionary[
+        "title_keywords"]
+
+
+@pytest.mark.django_db
+def test_create_new_job_form_with_invalid_data(example_recruiter, example_job_data_as_dictionary,
+                                               client):
+    example_job_data_as_dictionary['title'] = ""  # invalid title
+    client.force_login(example_recruiter.user)
+    response = client.post("/recruiter/create_new_job_form/", data=example_job_data_as_dictionary)
+    assert response.status_code == 200
```

Comment:
Do you get status code 200?

Reply:
Yea I do get status code 200 since the data invalid so I'm staying at the same page. I believe that's what you spoke about with Barak yesterday in the lecture